torch/_refs/__init__.py

```diff
@@ -2589,7 +2589,7 @@ def constant_pad_nd(
     if value == 0 and input.dtype == torch.bool:
         value = False
     # torch.fill isn't typed to allow complex values
-    output = torch.fill(output, value)  # type: ignore[arg-type]
+    output = output.fill_(value)  # type: ignore[arg-type]
```

Comment:
no need to assign it to output.

Reply:
Is there an out-of-place way to achieve this effect?

Reply:
The old code was done out of place.
I guess that fill() does not preserve the layout from the given input and thus cannot be used?
We do have `torch.full()` to do `torch.empty` + `fill_` though, maybe you want to use that?

Reply:
Seems like fill should be preserving strides:
```
In [1]: import torch

In [2]: a=torch.randn(3,4).t()

In [3]: b=torch.fill(a, 0)

In [4]: b.stride()
Out[4]: (1, 4)
```

Reply:
>>> a = torch.empty(3,3,3,3, memory_format=torch.channels_last, device='meta')
>>> a.stride()
(27, 1, 9, 3)
>>> b = torch.fill(a, 1)
>>> b.stride()
(27, 9, 3, 1)

This is a problem for meta tensor... aten.fill have incorrect stride for mete tensor

Reply:
Fixed meta function for aten.fill in this PR. 
BTW, we don't have opinfo coverage for aten.fill. :(

Reply:
Cool, that's better